File: DSAlgorithms/Leetcode/c++/arrays_hash/1299-replace_with_greatest.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
       int right_max = -1;
       for(int i = arr.size() - 1; i >= 0; i--)
       {
           int new_max = std::max(right_max, arr[i]);
           arr[i] = right_max;
           right_max = new_max;
       } 
       return arr;
    }

File: dsa_algo/leetcode/ts/src/arrays_hash/1299-replace_with_greatest.ts
const replaceElements = (arr: number[]): number[] => {
    let current_max = -1;
    for(let i = arr.length - 1; i >= 0; i --)
    {
        let new_max = Math.max(current_max, arr[i]);
        arr[i] = current_max;
        current_max = new_max;
    }
    return arr;
};
